redis/src/cluster_async/mod.rs

```diff
@@ -393,13 +392,15 @@ pub(crate) struct ClusterConnInner<C> {
     #[allow(clippy::complexity)]
     in_flight_requests: stream::FuturesUnordered<Pin<Box<Request<C>>>>,
     refresh_error: Option<RedisError>,
-    // A flag indicating the connection's closure and the requirement to shut down all related tasks.
-    shutdown_flag: Arc<AtomicBool>,
+    // Handler of the periodic topology check task.
+    topology_check_task: Option<tokio::task::JoinHandle<()>>,
 }
 
 impl<C> Dispose for ClusterConnInner<C> {
     fn dispose(self) {
-        self.shutdown_flag.store(true, Ordering::Relaxed);
+        self.topology_check_task
+            .map(|handle| handle.abort())
+            .unwrap_or_default();
```

Comment:
why do we need to unwrap it? 
and we should log a warning if we failed to stop the periodic checks

Reply:
dispose is part of Dispose trait and cant return any value

Reply:
also abort dont return any value

Reply:
oh it looks like abort doesn't return a result so to check it we need to check handle.await result, but it's an async call so we can't do that. Anyway seems like abort always succeed from their doc.

Reply:
Anyhow its a bit problematic, cause to who exactly you are going to return it? you dont call it.
We can log a warn if its a good sulotion.